lua/rocks/config/internal.lua

```diff
@@ -70,25 +70,59 @@ local default_config = {
         ---@type string[]
         unrecognized_configs = {},
     },
+    ---@type fun(path: string): string[]
+    get_imports = function(path)
+        local config_file = fs.read_or_create(path, constants.DEFAULT_CONFIG)
+        local rocks_toml = require("toml_edit").parse_as_tbl(config_file)
+        if rocks_toml.import ~= nil and type(rocks_toml.import) == "table" then
+            return rocks_toml.import
+        end
+        return {}
+    end,
     ---@type fun():RocksToml
     get_rocks_toml = function()
-        local config_file = fs.read_or_create(config.config_path, constants.DEFAULT_CONFIG)
-        local rocks_toml = require("toml_edit").parse_as_tbl(config_file)
-        for key, tbl in pairs(rocks_toml) do
-            if key == "rocks" or key == "plugins" then
-                for name, data in pairs(tbl) do
-                    if type(data) == "string" then
-                        ---@type RockSpec
-                        rocks_toml[key][name] = {
-                            name = name,
-                            version = data,
-                        }
-                    else
-                        rocks_toml[key][name].name = name
+        local rocks_toml = {}
+        local visited = {}
+
+        local function parse(file_path, default)
+            -- Don't allow recursive includes
+            if visited[file_path] then
+                return nil
+            end
+            visited[file_path] = true
+
+            local config_file = fs.read_or_create(file_path, default)
+            local rt = require("toml_edit").parse_as_tbl(config_file)
+
+            -- Setup rockspec for rocks/plugins
+            for key, tbl in pairs(rt) do
+                if key == "rocks" or key == "plugins" then
+                    for name, data in pairs(tbl) do
+                        if type(data) == "string" then
+                            ---@type RockSpec
+                            rt[key][name] = {
+                                name = name,
+                                version = data,
+                            }
+                        else
+                            rt[key][name].name = name
+                        end
                     end
                 end
             end
+            -- Merge, giving preference to existing config before imports
```

Comment:
Yeah, you're right. You'd want imports to have preference over the base config so that you could have local config override scm config if you wanted to do that. Since this is a valid use case, we may now want to raise an error. I'll change it to give preference to the imported config.

Reply:
The reason I'm not sure if we want to support overrides is because it increases the complexity: When updating plugins that exist in multiple configs, you have to keep track of which toml file to write to.
There might be other subtle things I'm not considering here. 

I'm not sure if the ability to override scm configs is strong enough of a use case, which is why I'd say YAGNI for now.

Reply:
I think (correct me if I'm wrong) because both the config and the operations respect config preferences, you will always write back to the config that has higher preference if it exists in both configs. The config.internal module merges all the configurations in order and the operations.helpers module orders the config tables by preference. So, I think it shouldn't add complexity but I may be wrong so I'll leave that up to you to decide.

Reply:
That's probably correct. But I'm still not sure if it has any implications I'm not foreseeing, which is why I'd rather not support overrides until people ask for it.

Reply:
come to think about it, we should probably add some tests for this feature.